### PR TITLE
Migrate most JavaScript code to Stimulus controllers

### PR DESCRIPTION
**Pull request by elia**
## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [ ] I have written a thorough PR description.
- [ ] I have kept my commits small and atomic.
- [ ] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

File: templates/app/views/products/_products-grid.html.erb [hunk 1]
```diff
@@ -1,14 +1,4 @@
...
-
-  # Optional props
-  classes = local_assigns.fetch(:classes, [])
-
-  # Classes
-  class_names = classes.push(base_class).join(" ")
-%>
-
-<ul class="<%= class_names %>">
+<ul class="products-grid">
```
Comment (waiting-for-dev) [inline templates/app/views/products/_products-grid.html.erb:1]: :heart:

File: templates/app/javascript/controllers/auto_submit.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
```
Comment (waiting-for-dev) [inline templates/app/javascript/controllers/auto_submit.js:3]: I think I'm missing something. Where is this controller used? :thinking:

**Review by waiting-for-dev — COMMENTED** [review]
Thanks, @elia! That's a great improvement for the experience with solidus starer frontend!

I left a question about the locale controller. Besides, I didn't manually test that all the behaviour remains in place, but I'm sure you did :slightly_smiling_face: 

Other than that, what do you think about splitting the work into different PRs? That would make reviewing and future references easier.

**Comment by fthobe** [comment]
@jarednorman Is this still viable?

**Comment by jarednorman** [comment]
Lots of conflicts. Unlikely this gets resurrected.

**Comment by fthobe** [comment]
Hey @elia , many of the smaller changes are still very valid. Would you mind if we pick them up? @jarednorman to reach compliance with structured data and ability to do google shopping the dynamic variant changes of the pdp need some changes and we can pick some of the stuff here up while we do that.

**Comment by elia** [comment]
@fthobe please do! Ruthlessly take or discard anything from this PR 🙌

**Comment by fthobe** [comment]
@elia what happens if somebody changes the currency in the storefront? 

Is that impacted?

https://github.com/solidusio/solidus_starter_frontend/commit/6932809a6119f67fe82e2455c04217ed39f6c942

**Comment by elia** [comment]
@fthobe I don't think so, at least in the commit you linked, except for bugs I think the behavior was expected to be the same for the whole PR.

**Pull request closed without merging**